CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/SyntaxBaseNodesFile.swift

```diff
@@ -35,117 +35,6 @@ let syntaxBaseNodesFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
       """
     )
 
-    DeclSyntax(
-      #"""
-      /// Extension of ``\#(node.kind.protocolType)`` to provide casting methods.
-      ///
-      /// These methods enable casting between syntax node types within the same
-      /// base node protocol hierarchy (e.g., ``DeclSyntaxProtocol``).
-      ///
-      /// While ``SyntaxProtocol`` offers general casting methods (``SyntaxProtocol.as(_:)``, 
-      /// ``SyntaxProtocol.is(_:)``, and ``SyntaxProtocol.cast(_:)``), these often aren't
-      /// appropriate for use on types conforming to a specific base node protocol
-      /// like ``\#(node.kind.protocolType)``. That's because at this level,
-      /// we know that the cast to another base node type (e.g., ``DeclSyntaxProtocol``
-      /// when working with ``ExprSyntaxProtocol``) is guaranteed to fail.
-      ///
-      /// To guide developers toward correct usage, this extension provides overloads
-      /// of these casting methods that are restricted to the same base node type.
-      /// Furthermore, it marks the inherited casting methods from ``SyntaxProtocol`` as 
-      /// deprecated, indicating that they will always fail when used in this context.
-      extension \#(node.kind.protocolType) {
-        /// Checks if the current syntax node can be cast to a given specialized syntax type.
-        ///
-        /// - Returns: `true` if the node can be cast, `false` otherwise.
-        public func `is`<S: \#(node.kind.protocolType)>(_ syntaxType: S.Type) -> Bool {
-          return self.as(syntaxType) != nil
-        }
-
-        /// Attempts to cast the current syntax node to a given specialized syntax type.
-        ///
-        /// - Returns: An instance of the specialized type, or `nil` if the cast fails.
-        public func `as`<S: \#(node.kind.protocolType)>(_ syntaxType: S.Type) -> S? {
-          return S.init(self)
-        }
-
-        /// Force-casts the current syntax node to a given specialized syntax type.
-        ///
-        /// - Returns: An instance of the specialized type.
-        /// - Warning: This function will crash if the cast is not possible. Use `as` to safely attempt a cast.
-        public func cast<S: \#(node.kind.protocolType)>(_ syntaxType: S.Type) -> S {
-          return self.as(S.self)!
-        }
-
-        /// Checks if the current syntax node can be upcast to its base node type (``\#(node.kind.syntaxType)``).
-        ///
-        /// - Returns: `true` since the node can always be upcast to its base node.
-        ///
-        /// - Note: This method overloads the general `is` method and is marked deprecated to produce a warning
-        ///         informing the user that the upcast will always succeed.
```

Comment:
I left the note's out. To me they read as notes to *us*, rather than developers using swift-syntax.